### PR TITLE
Stinger API

### PR DESCRIPTION
**Pull request by Owen1212055**
Adds stinger api, this is pretty much copied from arrows in body api.

### CODE REVIEW

File: patches/api/0320-Stinger-API.patch [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
++     * @param ticks time until stinger leaves
++     */
++    public void setStingerCooldown(int ticks);
++
++    /**
++     * Gets the amount of stingers in an entity's body.
++     *
++     * @return amount of stingers in body
++     */
++    public int getStingersInBody();
```
Comment (Trigary) [inline patches/api/0320-Stinger-API.patch]: I think `getStingerCount` would be a better, more descriptive method name. Same goes for the setter.

Comment (Owen1212055) [reply patches/api/0320-Stinger-API.patch]: Is it worth the inconsistency between arrows? Arrow methods call it arrows in body.

Comment (Trigary) [reply patches/api/0320-Stinger-API.patch]: Probably not. Apologies.

**Comment by Owen1212055** [comment]
rebased

**Review by Machine-Maker — APPROVED** [review]
Almost made the **exact** same comment that Trigary did about getStingersInBody.

File: patches/api/0323-Stinger-API.patch [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+index cda05df6784dd4d6a09710a416dcb71c016dabfc..06812f127e29b5c4cd4328a09b822cdb880ca4b3 100644
+--- a/src/main/java/org/bukkit/entity/LivingEntity.java
++++ b/src/main/java/org/bukkit/entity/LivingEntity.java
+@@ -336,6 +336,36 @@ public interface LivingEntity extends Attributable, Damageable, ProjectileSource
+      */
+     public void setArrowsInBody(int count);
+ 
++    // Paper Start - Stinger API
++    /**
++     * Gets the time in ticks until the next stinger leaves the entity's body.
```
Comment (Proximyst) [inline patches/api/0323-Stinger-API.patch]: What is a stinger? Add to the docs.

**Review by Proximyst — CHANGES_REQUESTED** [review]

File: patches/api/0323-Stinger-API.patch (hunk 2 above)
Comment (Owen1212055) [reply patches/api/0323-Stinger-API.patch]: image

Stingers appear in your body when you get stung by bees, similar to arrows. Should I just document it as bee stinger? Cause I am not sure.

Comment (Proximyst) [reply patches/api/0323-Stinger-API.patch]: Yeah, as simple as just adding `bee` in front of every `stinger` in this PR would definitely be enough; that should at least be more Googleable if it still doesn't click.

**Comment by Owen1212055** [comment]
Okay, it is now referred to as bee stinger.

**Review by Proximyst — APPROVED** [review]

**Pull request merged**